onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -330,6 +330,54 @@ TEST(InferenceSessionTests, DisableCPUArena) {
   RunModel(session_object, run_options);
 }
 
+TEST(InferenceSessionTests, TestModelSerialization) {
+  // Load model with level1 transform level as session options 
+  // and serialize the model after transformation.
```

Comment:
Does anything validate that the model was actually changed by the optimizations and differs from the original when saved?

Reply:
No, we are not validating if the model was optimized as that seems to be a unit test for graph transformer, and out of scope for this unit test. All the optimized_model_filepath option does is serialize the model what ever it may be.

Reply:
@skottmckay  Let me know if you think otherwise.

Reply:
I'm not sure that the graph transformer unit tests, which are isolated from the InferenceSession, are a good enough validation. My thinking was that to validate the saving of the model you need to a) change the model (otherwise Graph::ToGraphProto just returns the original proto); and b) do it via InferenceSession. The graph transformer tests cover (a) but not (b).